### ISSUE TITLE
lazy loading, many-to-one, querying on the wrong entity id.

### ISSUE THREAD

**Issue by ericwooley** [typeorm/typeorm#234]
***I updated the details below after further investigation***

I am having an issue where I am lazy loading a relationship, but a where an [the wrong entity id is being queried](https://github.com/ericwooley/graphql-typeorm-passport-boilerplate/blob/master/src/graphql/queries/user.ts#L19)

`executing query: SELECT "lists"."id" AS "lists_id", "lists"."name" AS "lists_name", "lists"."owner" AS "lists_owner" FROM "shopping_list" "lists" INNER JOIN "user" "User" ON "User"."id"="lists"."owner" WHERE "lists"."id"=$1 -- PARAMETERS: [1]`

Should just be 
`executing query: SELECT "lists"."id" AS "lists_id", "lists"."name" AS "lists_name", "lists"."owner" AS "lists_owner" FROM "shopping_list" "lists" INNER JOIN "user" "User" ON "User"."id"="lists"."owner" WHERE "user"."id"=$1 -- PARAMETERS: [1]`

If I don't lazy load and use add the join to the repo.findById (everything works fine)[https://github.com/ericwooley/graphql-typeorm-passport-boilerplate/blob/master/src/graphql/queries/user.ts#L6]

```
const user = await userRepo.findOneById(request.user.id, {
      alias: 'user',
      innerJoinAndSelect: {
        "lists": "user.lists"
    }
})
```

I have a [user](https://github.com/ericwooley/graphql-typeorm-passport-boilerplate/blob/master/src/models/user.ts#L26) class, with a lazy loaded one to many relationship: 
```
@Table()
export default class User {
	// ...
	@OneToMany(type => ShoppingList, shoppingList => shoppingList.owner, {
		cascadeAll: true
	})
	lists: Promise<ShoppingList[]>
        // ... 
}
```

and a class for [ShoppingList](https://github.com/ericwooley/graphql-typeorm-passport-boilerplate/blob/master/src/models/shoppingList.ts)
```
@Table()
       // ...
	@ManyToOne(type => User, user => user.lists, {
		cascadeAll: true,
	})
	owner: Promise<User>
}
```

I cannot figure out for the life of me, why this where clause is being added. Maybe it's something dumb I am doing, but I have tried messing with it every way I can think of.

**Comment by pleerock** [comment]
when you can `user.list` you want to load all lists of this SPECIFIC user. If your query will be:

`SELECT "lists"."id" AS "lists_id", "lists"."name" AS "lists_name", "lists"."owner" AS "lists_owner" FROM "shopping_list" "lists" INNER JOIN "user" "User" ON "User"."id"="lists"."owner"`

then it will load all lists of all users. That's why this id exist. Do you have problems with this query? Is result incorrect or something else?

**Comment by ericwooley** [comment]
@pleerock The problem is I am only getting one list because `WHERE "lists"."id"=$1 -- PARAMETERS: [1]` is being added. You are correct though, I mistakenly said it needs to have no where clause. What it needs is a where clause of `WHERE "User"."id" = 1` as thats the id of the owner.

 What I actually need is where User.id = 1

**Comment by Fokir** [comment]
It is not accidentally repeat my problem [https://github.com/typeorm/typeorm/issues/191](#191)?

**Comment by ericwooley** [comment]
@Fokir Thanks for pointing that out. I'll see if the alpha version fixes this later today.

**Comment by ericwooley** [comment]
@Fokir 

It does version 0.0.7-alpha.8 (or alpha.25, or 0.0.7, or 0.0.8) does not appear to fix my issue. I am going to update the issue now

**Comment by pleerock** [comment]
fixed and released in `0.0.8-alpha.2`
